subprojects/docs/src/docs/userguide/troubleshooting.adoc

```diff
@@ -15,29 +15,177 @@
+
+==== Permission denied
+
+If you get "permission denied", that means that Gradle likely exists in the correct place, but it is not executable.
+You can fix this using `chmod +x path/to/executable` on *nix-based systems.
+
+==== Other installation failures
+
+If `gradle --version` works, but all of your builds fail with the same error, it is possible there is a problem with one of your Gradle build configuration scripts.
+
+You can verify the problem is with Gradle scripts by running `gradle help` which executes configuration scripts, but no Gradle tasks. If the error persists, build configuration is problematic.
+If not, then the problem exists within the execution of one or more of the requested tasks (Gradle executes configuration scripts first, and then executes build steps).
+
+[[sec:troubleshooting_dependency_resolution]]
+=== Debugging dependency resolution
+
+Common dependency resolution issues such as resolving version conflicts are covered in <<troubleshooting_dependency_resolution>>.
+
+You can see a dependency tree and see which resolved dependency versions differed from what was requested by clicking the _Dependencies_ view and using the search functionality, specifying the resolution reason.
+
+++++
+<figure>
+    <title>Debugging dependency conflicts with build scans</title>
+    <imageobject>
+        <imagedata fileref="img/troubleshooting-dependency-management-build-scan.png"/>
+    </imageobject>
+</figure>
+++++
+
+The link:https://scans.gradle.com/s/sample/troubleshooting-userguide/dependencies?expandAll&filters=WzFd&toggled=W1swXSxbMF0sWzAsMF0sWzAsMV1d[actual build scan] with filtering criteria is available for exploration.
+
+[[sec:troubleshooting_performance]]
+=== Troubleshooting slow Gradle builds
+
+For build performance issues (including “slow sync time”), see the guide to link:https://guides.gradle.org/performance/[Improving the Performance of Gradle Builds].
+
+Android developers should watch a presentation by the Android SDK Tools team about link:https://youtu.be/7ll-rkLCtyk[Speeding Up Your Android Gradle Builds].
+Many tips are also covered in the Android Studio user guide link:https://developer.android.com/studio/build/optimize-your-build.html[on optimizing build speed].
+
+[[sec:troubleshooting_build_logic]]
+=== Debugging build logic
+
+==== Attaching a debugger to your build
+
+You can set breakpoints and debug <<sec:packaging_a_plugin,buildSrc and standalone plugins>> in your Gradle build itself by setting the `org.gradle.debug` property to “true” and then attaching a remote debugger to port 5005.
+
+----
+❯ gradle help -Dorg.gradle.debug=true --no-daemon
```

Comment:
@eriwen Why the `--no-daemon`? `-Dorg.gradle.debug=true` makes sure the debugging arguments are passed to the daemon, so a new daemon is started if there is no compatible daemon around.
You can then directly connect to the daemon on port `5005`.

Reply:
@wolfs When I was trying this out a bunch, it was not my experience that a new Daemon was started (though it should've been). So the build would run using the existing Daemon and not break. `--no-daemon` was the only way I could reliably get debugging working (at least for Kotlin scripts)

Reply:
Not everytime a new daemon is started. If a daemon running already with the debugging flags exists, then it will be re-used. But you should already be able to connect to that daemon without starting the Gradle build since it has debugging enabled.

The build would not wait for you to connect, since the daemon already waited the first time and you disconnected.

If this does not work (using `-Dorg.gradle.debug=true` with a daemon), then it should be considered a bug.